service/storage_service.cc

```diff
@@ -5053,7 +5053,17 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(raft
                                 version, current_version)));
                     }
 
+                    seastar::timer<seastar::lowres_clock> stall_timer;
+                    auto start = std::chrono::steady_clock::now();
+                    stall_timer.set_callback([&] {
+                        ss._shared_token_metadata.mark_stalled(start);
+                    });
+                    auto threshold = ss._db.local().get_config().topology_barrier_stall_detector_threshold_seconds();
+                    stall_timer.arm(std::chrono::duration_cast<seastar::lowres_clock::duration>(
+                                std::chrono::duration<double>(threshold)));
                     co_await ss._shared_token_metadata.stale_versions_in_use();
+                    stall_timer.cancel();
```

Comment:
What happens if we skip this by exception? I guess timer's destructor will cancel the pending call.

Reply:
Yes, it will be canceled.